### PR TITLE
Add touch-action:manipulation on interactive elements to avoid 300ms click delay

### LINKED ISSUE #18044: Add touch-action:manipulation to future-proof avoidance of 300ms click delay?
In particular, the next iOS Safari feature release should (modulo secret Apple machinations) support it (see http://trac.webkit.org/changeset/191452), whereas it's uncertain whether WebKit will (ever) decide to remove the delay on all pages with mobile viewports (see https://bugs.webkit.org/show_bug.cgi?id=150469) like Chrome et al. already do.
Some background on the delay: http://developer.telerik.com/featured/300-ms-click-delay-ios-8/
CC: Touchy guy @patrickhlauke

### PR DESCRIPTION
**Pull request by cvrebert**
Fixes #18044.

### CODE REVIEW

**Comment by slavanga** [comment]
The touch-action property on .btn is no longer needed after this gets merged.
https://github.com/twbs/bootstrap/blob/v4-dev/dist/css/bootstrap.css#L2208

File: scss/_reboot.scss [hunk 1]
```diff
@@ -204,6 +204,19 @@ img {
...
+// In particular, this should be supported in the next post-iOS-9.1 Safari feature release (see http://trac.webkit.org/changeset/191452 )
+
+a,
+area,
+button,
+[role="button"],
+input,
+label,
+select,
+summary {
```
Comment (patrickhlauke) [inline scss/_reboot.scss]: maybe add `textarea` too?

**Comment by patrickhlauke** [comment]
@slavanga yup, that would make sense.

apart from that (and adding `textarea`) this LGTM

**Comment by cvrebert** [comment]
Applied both of the suggested improvements.

**Comment by patrickhlauke** [comment]
:shipit:

**Comment by cvrebert** [comment]
CC: @mdo

**Comment by cvrebert** [comment]
Hmm, looks like Apple has jumped on the "kill the delay when `width=device-width`" bandwagon:
http://trac.webkit.org/changeset/191644

I guess this would still be useful for mobile IE10-11 (and Edge on laplets? Am I reading your table right @patrickhlauke ?)

**Comment by cvrebert** [comment]
@patrickhlauke Can you confirm that Edge laplet thing?

**Comment by patrickhlauke** [comment]
sorry @cvrebert this one slipped me by. you're correct, as per http://patrickhlauke.github.io/touch/tests/results/#suppressing-300ms-delay IE/Edge on Win 10 laptop/tablet/Surface with touchscreen does have the 300ms delay, ignores viewport meta, but does react to `touch-action`. in short, yes still useful for that scenario.

**Comment by cvrebert** [comment]
Thanks. I'll update the comment in the code accordingly.

**Comment by mdo** [comment]
Neat! :shipit:

**Comment by mdo** [comment]
We should add some docs for this though to explain it.

**Comment by cvrebert** [comment]
Updated the comment. Just have to write some docs now.

**Comment by patrickhlauke** [comment]
@cvrebert if you have an idea where in the docs it should go, I can help/have a go at it.

**Comment by cvrebert** [comment]
@patrickhlauke Pretty sure it belongs in https://github.com/twbs/bootstrap/blob/v4-dev/docs/content/reboot.md

**Comment by patrickhlauke** [comment]
ok, leave it with me, i'll draft something up...after all, i AM "uncle touchy"...

**Comment by cvrebert** [comment]
Docs for this will be covered by #18266.

**Pull request merged**